test/tools/ossfuzz/strictasm_diff_ossfuzz.cpp

```diff
@@ -70,6 +70,8 @@ extern "C" int LLVMFuzzerTestOneInput(uint8_t const* _data, size_t _size)
 	ostringstream os1;
 	ostringstream os2;
 	yulFuzzerUtil::interpret(os1, stack.parserResult()->code);
+	if (os1.str().find("Interpreter execution step limit reached.") != std::string::npos)
```

Comment:
Instead of hard-coding a string-find on an error message, can we return a status code instead that is reflecting the success and failure type (such as this)? This frees you up from bugs because of foreign string changes and makes the test a little more save (same below).

Reply:
Return codes other than zero are reserved by the fuzzer so I can't use them here.

Reply:
But `yulFuzzerUtil::interpret` could return a code - couldn't it?

Reply:
Actually, I'm not sure how to use return codes to signal failure in our current architecture. Right now, we throw an exception inside the interpreter when the first of the following happens
  - reached maxSteps (this PR)
  - reached trace size limit
Before throwing the exception, the interpreter appends informative text to the trace for debug purposes.

These exceptions are then caught by `yulFuzzerUtil::interpret` simply relays the trace to the caller i.e., the fuzzing harness. That is the reason I "grep" for it in the harness.

Reply:
You can either check for `m_state.numSteps >= m_state.maxSteps` or throw a different exception in that case. It actually might make sense to throw a different exceptions (where both exceptions share a comment base class) in the "exceptional" termination cases.

Reply:
> Please change `yulFuzzerUtil::interpret` to return a status code.

I'm confused, status code and throwing exceptions (previous comment) are two different approaches :) Which one?

Reply:
Right! Whatever is more convenient :)

Reply:
If you use an exception, you can of course also just catch it outside.

Reply:
I used the following approach. Created a flat exception hierarchy that subclasses `dev:Exception` with the following exceptions specific to the yulInterpreter
  - `ExplicitlyTerminated`: triggered during the execution of explicitly terminating opcodes such as `stop`, `revert`, `return` etc.
  - `StepLimitReached`: triggered when the interpreter has evaluated more than `stepLimit` blocks
  - `TraceLimitReached`: triggered when the interpreter has created a trace whose size exceeds the `traceLimit`

Updated exception handling in `yulrun` and the fuzzer code to reflect this change.